android/quest/src/main/java/org/smartregister/fhircore/quest/ui/usersetting/UserSettingViewModel.kt

```diff
@@ -228,4 +239,63 @@ constructor(
   suspend fun emitSnackBarState(snackBarMessageConfig: SnackBarMessageConfig) {
     _snackBarStateFlow.emit(snackBarMessageConfig)
   }
+
+  private fun copyDatabase(context: Context, onCopyCompleteListener: () -> Unit) {
+    viewModelScope.launch(dispatcherProvider.io()) {
+      try {
+        val passphrase = DBUtils.getEncryptionPassphrase("fhirEngineDbPassphrase")
+
+        val dbFilename = if (BuildConfig.DEBUG) "resources" else "resources_encrypted"
+        val dbFile = File("/data/data/${context.packageName}/databases/$dbFilename.db")
+
+        val downloadsDir =
+          Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS)
+
+        val username = secureSharedPreference.retrieveSessionUsername()
+        val practitionerId =
+          sharedPreferencesHelper.read(SharedPreferenceKey.PRACTITIONER_ID.name, username)
+
+        val backupFile =
+          File(
+            downloadsDir,
+            String.format(
+              "%s_%s_%s_%s.db",
+              applicationConfiguration.appTitle.replace(" ", "_"),
+              username,
+              practitionerId,
+              today().formatDate("yyyyMMdd-HHmmss"),
+            ),
+          )
+
+        val dbCopied =
+          if (BuildConfig.DEBUG) {
+            DBUtils.copyUnencryptedDb(dbFile, backupFile)
+          } else {
+            DBUtils.decryptDb(dbFile, backupFile, passphrase)
+          }
+
+        if (dbCopied) {
+          // The password is currently set as:
+          // {username}_{first 8 characters of the practitioner_id}
+          val zipFile = File("${backupFile.absolutePath}.zip")
+          FileUtils.zipFiles(
+            zipFile,
+            listOf(backupFile),
+            String.format(
+              "%s_%s",
+              username,
+              practitionerId!!.substring(0, practitionerId.indexOf("-")),
+            ),
```

Comment:
Extract this parameter to a variable to improve readability.  Can you use String interpolation instead?